components/apps/stream/VestingElement.tsx

```diff
@@ -145,7 +153,7 @@ export function VestingElement({vesting}: {vesting: TStreamArgs}): ReactElement
 				<div className={'w-full md:w-auto'}>
 					<Button
 						onClick={onClaim}
-						isDisabled={!address}
+						isDisabled={!address || (!vesting.open_claim && address !== vesting.recipient)}
```

Comment:
that's a nitpick, but it's probably better to move this logic into separate variables for better readability